stellargraph/core/convert.py

```diff
@@ -150,3 +154,144 @@ def convert_edges(
     assert all(features is None for features in edge_features.values())
 
     return EdgeData(edges, node_data)
+
+
+NodeInfo = namedtuple("NodeInfo", ["ids", "features"])
+
+
+def _empty_node_info() -> NodeInfo:
+    return NodeInfo([], [])
+
+
+def _features_from_attributes(node_type, num_nodes, values, dtype):
+    # the size is the first element that has a length, or None if there's only None elements.
+    size = next((len(x) for x in values if x is not None), None)
+
+    if size is None:
+        # no features = zero-dimensional features, and skip the loop below
+        return np.empty((num_nodes, 0), dtype)
+
+    default_value = np.zeros(size, dtype)
+
+    def compute_value(x):
+        if x is None:
+            return default_value
```

Comment:
Yeah. This actually differs slightly to the NetworkX StellarGraph, where it would just leave nodes out of the feature matrices, and queries for their features would fail. It was inspired by the behaviour of querying for a node that doesn't exist (e.g. an element of a GraphSAGE/HinSAGE sample for which there's no neighbour).

Another approach would be to fail fast here and throw an exception. What do you think?

Reply:
I think I prefer this behaviour over the NetworkX StellarGraph, as long as the user is informed about it. Maybe instead of an exception, we could also print out a warning like `f"found {missing} nodes with missing features - using zeros"`? I think we want to help the user with a solution when they have missing features, and filling it with zeros seems like a reasonable approach.

Reply:
Done